src/main/java/recipeditor/parser/FlagParser.java

```diff
@@ -1,64 +1,73 @@
 package recipeditor.parser;
 
+import recipeditor.exception.ExcessFlagsException;
 import recipeditor.recipe.Recipe;
 
 import java.util.ArrayList;
 
-public interface FlagParser {
+import static recipeditor.parser.FlagType.*;
+
+public class FlagParser {
 
     /**
      * Fina all the flags in a given command.
      *
      * @param parsedCommand the parsed command which to find flags from
      * @return the list of flags found in command
      */
-    static FlagType[] getFlags(String[] parsedCommand) {
-        FlagType[] flags = new FlagType[2];
-        int commandFlagCount = 0;
-        int recipeFlagCount = 0;
-        for (String s: parsedCommand) {
+    public static FlagType[] getFlags(String[] parsedCommand) throws ExcessFlagsException {
+        FlagType[] flags = {NULL, NULL};
+        int recipeFlagCount = 0, commandFlagCount = 0;
+        for (String s : parsedCommand) {
             if (s.contains("-")) {
                 switch (s) {
+                case "-id":
```

Comment:
This seems to be a magic literal. Maybe you want to use a named constant so that the code is easier to understand. This

Reply:
This issue has been solved in my PR of #164. Also, there seems to have conflicting files due to the latest merging of branch. You may want to try resolving the conflicts first.

Reply:
ok, all magic literals are replaced aligning to #164 after resolve conflicts